apps/designer/app/designer/features/props-panel/use-props-logic.ts

```diff
@@ -127,9 +125,39 @@ export const usePropsLogic = ({
       const isPropRequired = draft[index].required;
 
       if (isPropRequired !== true) {
-        draft[index].prop = name;
-        draft[index].value = defaultValue;
+        // @todo we need to now allow to change non required on required prop too
+        draft[index] = {
+          id: draft[index].id,
+          prop: name,
+          value: defaultValue,
+        };
+      }
+    })(userProps);
+
+    // Optimistically update the state (what if publish fails?)
+    setUserProps(nextUserProps);
+
+    const updatedProps = nextUserProps.filter((item) => item.id === id);
+
+    updateProps(updatedProps);
+  };
+
+  const handleChangePropValue: HandleChangePropValue = (id, value) => {
+    const nextUserProps = produce((draft: Array<UserProp>) => {
+      const index = draft.findIndex((item) => item.id === id);
+      const val = draft[index];
```

Comment:
instead of using abbreviations just to not have a namespace conflict, we can use a more specific names, e.g. 
```suggestion
      const draftValue = draft[index];
```